addons/mail/static/src/models/emoji_registry.js

```diff
@@ -31,19 +44,15 @@ registerModel({
             if (!this.exists()) {
                 return;
             }
-            await this.messaging.executeGracefully(dataEmojis.map(emojiData => () => {
```

Comment:
It's best to keep the gracefully, so that it doesn't keep main thread while inserting all the emojis

Reply:
Just for context, the fact that the inserts are all individual + executeGracefully makes it so that the inserts take ~8 seconds before the change, which is why I changed it to a batched insert, which takes ~800ms. If we don't want to block the main thread for 800ms, it's still possible to use smaller batches.

Reply:
Ok, this graceful and slow execution was essential for good working of tours when emoji data were populated at page load (e.g. maintaining main thread for too long prevented bootstrap animation like drop-down).

Now that's lazy y loaded by a user interaction, it's "alright" when blocking is lower than 1sec. Of course we'll find ways to improve performance, but right now it's the best decision to make it faster 🙂